src/MLDatasets.jl

```diff
@@ -2,28 +2,35 @@ module MLDatasets
 
 using FixedPointNumbers
 using SparseArrays
-using DataFrames, Tables
-using Glob
-import ImageCore
-using ColorTypes
+using Tables
+using DataDeps
 import MLUtils
 using MLUtils: getobs, numobs, AbstractDataContainer
-
-### I/O imports
-import NPZ
-import Pickle
-using MAT: matopen, matread
-import CSV 
-using HDF5
-using JLD2
-import JSON3
+using ColorTypes
+using Glob
 using DelimitedFiles: readdlm
-##########
+using FileIO
+using LazyModules: @lazy
 
-export getobs, numobs
-export convert2image
+include("require.jl") # export @require
 
+# Use `@lazy import SomePkg` whenever the returned types are not its own types,
+# since for methods applied on the returned types we would encounter in world-age issues
+# (see discussion in  https://github.com/JuliaML/MLDatasets.jl/pull/128).
+# In the other case instead, use `require import SomePkg` to force 
+# the use to manually import.
 
+@require import JSON3="0f8b85d8-7281-11e9-16c2-39a750bddbf1"
+@require import DataFrames="a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+@lazy import ImageCore="a09fc81d-aa75-5fe9-8630-4744c3626534"
```

Comment:
FWIW, I believe we need to eagerly import `Colors` and `ColorVectorSpace` to properly avoid the world-age issue brought by lazily importing `ImageCore`.

Reply:
Ok, so we can just make ImageCore a `@require`. Or  even better, we could `@require import ImageShow`, and from there we get `ImageShow.ImageCore` and `ImageShow.ImageCore.RGB` and nice display in editors.

Reply:
Or even the entire Images https://github.com/JuliaImages/Images.jl/issues/1009, if you'd say so.

But again I'm not very confident that `@require` is a good direction, let's stick to `ImageCore` only for this PR.

Reply:
but ImageShow doesn't add much much on top of ImageCore and it is something that most likely people will import in any case. Having an error message that helps discovering ImageShow is useful (it would have been useful to me https://github.com/fonsp/Pluto.jl/discussions/2097)

Reply:
> but ImageShow doesn't add much much on top of ImageCore and it is something that most likely people will import in any case.

This entirely depends on whether you're a user or a developer. If you're a user, we try to make `using Images` the simple answer (and #1009 to lazify the entire Images). If you're a developer, there's no need to depend on ImageShow -- no matter how small it is.